src/main/scala/rps/Game.scala

```diff
@@ -5,66 +5,92 @@ import scala.util.Random
 
 object Game {
 
-  // TODO this should be an enum (or Scala equivalent of TS Enum)
-  object Weapon {
-    val Rock: String = "0" // 🪨
-    val Paper: String = "1" // 📄
-    val Scissors: String = "2" // ✂️
+  sealed trait Move
+  object Move {
+    case object Rock extends Move // 🪨
+    case object Paper extends Move // 📄
+    case object Scissors extends Move // ✂️
+    val moves = List(Rock, Paper, Scissors)
+
+    // Idea is similar to a fp-ts Encoder, I dunno if it's the right place here
+    def decode(input: String): Option[Move] = {
+      input match {
+        case "0" => Some(Move.Rock)
```

Comment:
```suggestion
        case "0" => Some(Rock)
```
Non serve `Move.` finché sei dentro l'`object`, per semplicità toglierei qui e sotto dove lo usi

Reply:
Incorporato 🔀